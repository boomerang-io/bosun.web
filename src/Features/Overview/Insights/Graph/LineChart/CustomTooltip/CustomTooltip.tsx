import React, { Component } from "react";
import "./styles.scss";

type Props = {
  payload?: any[];
};

class CustomTooltip extends Component<Props> {
  renderTooltip = (tooltipFields: any) => {
    return tooltipFields.map((field: any, index: any) => {
      return (
        <p key={`${field}-${index}`} className="b-tooltip--data">
          {" "}
          <span className="b-tooltip--field">{`${
            field === "passRate" ? "pass rate" : field === "criticalViolations" ? "critical violations" : field
          }: `}</span>
          {this.props.payload && `${this.props.payload[0].payload[field]}`}
        </p>
      );
    });
  };
  render() {
    const tooltipFields =
      this.props.payload && this.props.payload.length > 0 ? Object.keys(this.props.payload[0].payload) : null;
    return (
      <div className="c-tooltip">
        {this.props.payload && this.props.payload.length > 0 ? <div>{this.renderTooltip(tooltipFields)}</div> : null}
      </div>
    );
  }
}

export default CustomTooltip;
