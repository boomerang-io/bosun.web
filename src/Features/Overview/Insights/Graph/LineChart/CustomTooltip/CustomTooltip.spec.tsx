import React from "react";
import CustomTooltip from "./index";
const payload = [
  {
    color: "#047cc0",
    dataKey: "total",
    fill: "url(#blueGradient)",
    fillOpacity: 0.6,
    formatter: undefined,
    name: "total",
    payload: { date: 1542160248412, failed: 5, success: 12, total: 17 },
    stroke: "#047cc0",
    unit: undefined,
    value: 17,
  },
];
describe("CustomTooltip --- Snapshot", () => {
  it("Capturing Snapshot of CustomTooltip", async () => {
    const { baseElement } = (global as any).rtlRouterRender(<CustomTooltip payload={payload} />);
    expect(baseElement).toMatchSnapshot();
  });
});
