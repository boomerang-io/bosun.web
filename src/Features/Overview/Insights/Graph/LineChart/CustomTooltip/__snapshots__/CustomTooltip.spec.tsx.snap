// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip --- Snapshot Capturing Snapshot of CustomTooltip 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="c-tooltip"
    >
      <div>
        <p
          class="b-tooltip--data"
        >
           
          <span
            class="b-tooltip--field"
          >
            date: 
          </span>
          1542160248412
        </p>
        <p
          class="b-tooltip--data"
        >
           
          <span
            class="b-tooltip--field"
          >
            failed: 
          </span>
          5
        </p>
        <p
          class="b-tooltip--data"
        >
           
          <span
            class="b-tooltip--field"
          >
            success: 
          </span>
          12
        </p>
        <p
          class="b-tooltip--data"
        >
           
          <span
            class="b-tooltip--field"
          >
            total: 
          </span>
          17
        </p>
      </div>
    </div>
  </div>
</body>
`;
